backend/inspirehep/records/api/mixins.py

```diff
@@ -203,17 +232,106 @@ def update_refs_in_citation_table(self, save_every=100):
                         citer_id=self.model.id,
                         cited_id=reference,
                         citation_date=citation_date,
+                        citation_type=RecordCitationType.citation,
                     )
                 )
             if len(references_waiting_for_commit) >= save_every:
                 db.session.bulk_save_objects(references_waiting_for_commit)
                 references_waiting_for_commit = []
         if references_waiting_for_commit:
             db.session.bulk_save_objects(references_waiting_for_commit)
+
+        _start = datetime.now()
+        if current_app.config.get("FEATURE_FLAG_ENABLE_SELF_CITATIONS"):
+            LOGGER.info("Starting self citations check")
+
+            self.update_self_citations()
         LOGGER.info(
             "Record citations updated",
             recid=current_record_control_number,
             uuid=str(self.id),
+            self_citations_update=(datetime.now() - _start).total_seconds(),
+        )
+
+    def get_authors_bais(self):
+        authors_list = []
+        for author in self.get_value("authors.ids", []):
+            for entry in author:
+                if entry.get("schema") == "INSPIRE BAI":
+                    authors_list.append(entry["value"])
+        return authors_list
+
+    def get_collaborations_values(self):
+        collaboration_list = []
+        for collaborations_values in self.get_value("collaborations.value", []):
+            for collaboration in collaborations_values:
+                collaboration_list.append(collaboration)
+        return collaboration_list
+
+    def get_self_cited_referenced_papers(self):
+        uuid = str(self.model.id)
+        sql_query = f"""
+            SELECT papers_from_authors.record_id
+            FROM
+                (SELECT DISTINCT a1.record_id FROM {RecordsAuthors.__tablename__} a1, {RecordsAuthors.__tablename__} a2
+                WHERE
+                    a2.record_id ='{uuid}'
```

Comment:
You mean table names or UUID?
Table names are defined by us, user never have access to them so no chance that names will be replaced, UUID is also internal thing, verified by Postgres. it's not possible in normal circumstances that user will pass malicious string to perform SQL injection in there.
Maybe if there will be some bug which allows user to modify memory in python interpreter, but then we would have bigger problems than SQL injection...
Do i miss something?

Reply:
I mean UUIDs mostly. Currently there's probably no way a user can inject a user id, but
1. you still have to think about it and prove that's indeed the case
2. you don't know how things might evolve, and if it's safe now doesn't mean it will remain safe forever.

On the other hand, if you use `text`, you know you can't have SQL injections, so I really don't understand why we're not doing the correct thing and make this query obviously safe (following the principle of [defense in depth](https://en.wikipedia.org/wiki/Defense_in_depth_(computing)).

Reply:
:+1: